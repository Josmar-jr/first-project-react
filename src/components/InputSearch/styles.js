import styled from "styled-components";

export const InputMain = styled.input`
  font-size: 1.6rem;
  font-weight: bold;

  width: 100%;

  margin: 2rem 0;
  padding: 0.2rem 1rem;
  border-radius: 8px;
`
