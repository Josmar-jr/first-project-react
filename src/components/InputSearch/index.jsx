import React from 'react';
import { InputMain } from './styles';

import P from 'prop-types';

// import { Container } from './styles';

export default function InputSearch({ handleChange, searchValue }) {

  return (
    <InputMain
      type="search"
      onChange={handleChange}
      value={searchValue}
      placeholder="Do your research"
    />
  )
}

InputSearch.propTypes = {
  searchValue: P.string.isRequired,
  handleChange: P.func.isRequired
}

