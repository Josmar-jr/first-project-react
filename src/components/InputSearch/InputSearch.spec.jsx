import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import InputSearch from ".";

describe('<InputSearch />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();

    render(<InputSearch handleChange={fn} searchValue={'test'} />)

    const input = screen.getByPlaceholderText(/Do your research/i)
    expect(input).toBeInTheDocument();

    expect(input.value).toBe('test');
  })

  it('should have a value of searchValue', () => {
    const fn = jest.fn();

    render(<InputSearch handleChange={fn} />)

    const input = screen.getByPlaceholderText(/Do your research/i)

    const value = 'value test'

    userEvent.type(input, value)

    expect(input.value).toBe('value test')
    expect(fn).toHaveBeenCalledTimes(value.length)
  })

  it('should match snapshot', () => {
    const fn = jest.fn();

    const { container } = render(<InputSearch handleChange={fn} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
