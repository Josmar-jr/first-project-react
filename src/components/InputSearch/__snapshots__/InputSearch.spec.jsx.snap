// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSearch /> should match snapshot 1`] = `
<input
  class="sc-bdnxRM fsJqKP"
  placeholder="Do your research"
  type="search"
  value=""
/>
`;
